snapcraft/meta/appstream.py

```diff
@@ -184,7 +194,14 @@ def _get_urls_from_xml_element(nodes, url_type) -> Optional[List[str]]:
             and node.attrib["type"] == url_type
             and node.text.strip() not in urls
         ):
-            urls.append(node.text.strip())
+            link = node.text.strip()
+            if validators.url(link) or validators.email(link):
+                urls.append(link)
+            else:
+                emit.progress(
+                    f"Ignoring invalid url {link!r} in {url_type!r} from appstream metadata.",
```

Comment:
This message could be slightly misleading if the user intended to use an email address, but it's not incorrect.